bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/HttpServiceUtil.java

```diff
@@ -39,45 +39,44 @@ public static int getHttpServicePortSecure(final BundleContext bc) {
     }
 
     // Utility method that could be used for non-secure and secure port.
-    @SuppressWarnings("rawtypes")
     private static int getHttpServicePortProperty(final BundleContext bc, final String propertyName) {
         Object value;
         int port = -1;
 
         // Try to find the port by using the service property (respect service ranking).
+        ServiceReference<?>[] refs = new ServiceReference[0];
         try {
-            int candidate = Integer.MIN_VALUE;
-            final ServiceReference[] refs = bc.getAllServiceReferences("org.osgi.service.http.HttpService", null);
-            if (refs != null) {
-                for (final ServiceReference ref : refs) {
-                    value = ref.getProperty(propertyName);
-                    if (value == null) {
-                        continue;
-                    }
-                    final int servicePort;
-                    try {
-                        servicePort = Integer.parseInt(value.toString());
-                    } catch (final NumberFormatException ex) {
-                        continue;
-                    }
-                    value = ref.getProperty(Constants.SERVICE_RANKING);
-                    final int serviceRanking;
-                    if (value == null || !(value instanceof Integer)) {
-                        serviceRanking = 0;
-                    } else {
-                        serviceRanking = (Integer) value;
-                    }
-                    if (serviceRanking >= candidate) {
-                        candidate = serviceRanking;
-                        port = servicePort;
-                    }
-                }
-            }
+            refs = bc.getAllServiceReferences("org.osgi.service.http.HttpService", null);
         } catch (final InvalidSyntaxException ex) {
             // This point of code should never be reached.
             final Logger logger = LoggerFactory.getLogger(HttpServiceUtil.class);
```

Comment:
As you have already pointed that this code block will never be executed as there is no OSGi filter associated, we can leave it empty. What do you think?